src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.spec.ts

```diff
@@ -147,4 +153,13 @@ describe('ServiceDaemonListComponent', () => {
   it('should not display doc panel if orchestrator is available', () => {
     expect(component.showDocPanel).toBeFalsy();
   });
+
+  it('should call daemon action', () => {
+    let daemon = daemons[0];
+    component.selection.selected = [daemon]
+    for(let action in ['start', 'stop', 'restart', 'redeploy']) {
+      component.daemonAction(action)
+      expect(daemonService.action).toHaveBeenCalledWith(daemon.daemon_name, action);
```

Comment:
I leave you the changes for the current tests to pass. Bear in mind that the last tests should also test that notificationService is also called:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.spec.ts
index 22e9354ca28..ca798848287 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.spec.ts
@@ -3,6 +3,7 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import _ from 'lodash';
 import { NgxPipeFunctionModule } from 'ngx-pipe-function';
+import { ToastrModule } from 'ngx-toastr';
 import { of } from 'rxjs';
 
 import { CephModule } from '~/app/ceph/ceph.module';
@@ -13,12 +14,10 @@ import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data
 import { SharedModule } from '~/app/shared/shared.module';
 import { configureTestBed } from '~/testing/unit-test-helper';
 import { ServiceDaemonListComponent } from './service-daemon-list.component';
-import { DaemonService } from '~/app/shared/api/daemon.service';
 
 describe('ServiceDaemonListComponent', () => {
   let component: ServiceDaemonListComponent;
   let fixture: ComponentFixture<ServiceDaemonListComponent>;
-  let daemonService: DaemonService;
 
   const daemons = [
     {
@@ -111,7 +110,14 @@ describe('ServiceDaemonListComponent', () => {
   };
 
   configureTestBed({
-    imports: [HttpClientTestingModule, CephModule, CoreModule, NgxPipeFunctionModule, SharedModule]
+    imports: [
+      HttpClientTestingModule,
+      CephModule,
+      CoreModule,
+      NgxPipeFunctionModule,
+      SharedModule,
+      ToastrModule.forRoot()
+    ]
   });
 
   beforeEach(() => {
@@ -156,10 +162,11 @@ describe('ServiceDaemonListComponent', () => {
 
   it('should call daemon action', () => {
     let daemon = daemons[0];
-    component.selection.selected = [daemon]
-    for(let action in ['start', 'stop', 'restart', 'redeploy']) {
-      component.daemonAction(action)
-      expect(daemonService.action).toHaveBeenCalledWith(daemon.daemon_name, action);
+    component.selection.selected = [daemon];
+    component['daemonService'].action = jest.fn(() => of());
+    for (let action in ['start', 'stop', 'restart', 'redeploy']) {
+      component.daemonAction(action);
+      expect(component['daemonService'].action).toHaveBeenCalledWith(daemon.daemon_name, action);
     }
   });
 });
```